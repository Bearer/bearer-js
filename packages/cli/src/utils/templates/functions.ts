import { Authentications } from '@bearer/types/lib/authentications'
import FunctionType from '@bearer/types/lib/function-types'
import * as Case from 'case'
import * as path from 'path'

import { copyFiles } from '../helpers'
import baseCommand from '../../base-command'

export default async (command: baseCommand, auth: Authentications, type: FunctionType, name: string) => {
  const vars = getVars(name, auth, type)
  return await copyFiles(command, path.join(`generate/function`, auth, type), command.locator.srcFunctionsDir, vars)
}

function getVars(name: string, authType: Authentications, functionType: FunctionType) {
  return {
    authType,
    functionType,
    fileName: name,
    functionClassName: Case.pascal(name)
  }
}
