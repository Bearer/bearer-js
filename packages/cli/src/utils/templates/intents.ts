import { Authentications } from '@bearer/types/lib/authentications';
import IntentType from '@bearer/types/lib/intent-types';
import * as Case from 'case';
import * as path from 'path';

import {copyFiles} from '../helpers'
import baseCommand from '../../base-command';


export default async (command: baseCommand, auth: Authentications, type: IntentType, name: string ) => {
  const vars = getVars(name,  auth, type)
  return await copyFiles(command, path.join(`generate/intent`, auth, type), command.locator.srcIntentsDir, vars)
}

function getVars(name: string, authType: Authentications, intentType: IntentType,) {
  return {
    authType,
    intentType,
    fileName: name,
    intentClassName: Case.pascal(name)
  }
}