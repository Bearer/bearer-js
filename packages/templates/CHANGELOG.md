# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.84.0"></a>
# [0.84.0](https://github.com/Bearer/bearer/compare/v0.83.5...v0.84.0) (2018-12-21)

**Note:** Version bump only for package @bearer/templates





<a name="0.83.4"></a>
## [0.83.4](https://github.com/Bearer/bearer/compare/v0.83.3...v0.83.4) (2018-12-21)

**Note:** Version bump only for package @bearer/templates





<a name="0.83.3"></a>
## [0.83.3](https://github.com/Bearer/bearer/compare/v0.83.2...v0.83.3) (2018-12-21)

**Note:** Version bump only for package @bearer/templates





<a name="0.83.0"></a>
# [0.83.0](https://github.com/Bearer/bearer/compare/v0.82.0...v0.83.0) (2018-12-20)

**Note:** Version bump only for package @bearer/templates





<a name="0.82.0"></a>
# [0.82.0](https://github.com/Bearer/bearer/compare/v0.81.9...v0.82.0) (2018-12-19)

**Note:** Version bump only for package @bearer/templates





<a name="0.81.7"></a>
## [0.81.7](https://github.com/Bearer/bearer/compare/v0.81.6...v0.81.7) (2018-12-19)

**Note:** Version bump only for package @bearer/templates





<a name="0.81.6"></a>
## [0.81.6](https://github.com/Bearer/bearer/compare/v0.81.5...v0.81.6) (2018-12-19)

**Note:** Version bump only for package @bearer/templates





<a name="0.81.3"></a>
## [0.81.3](https://github.com/Bearer/bearer/compare/v0.81.2...v0.81.3) (2018-12-18)

**Note:** Version bump only for package @bearer/templates





<a name="0.81.0"></a>
# [0.81.0](https://github.com/Bearer/bearer/compare/v0.80.1...v0.81.0) (2018-12-17)

**Note:** Version bump only for package @bearer/templates





<a name="0.80.0"></a>
# [0.80.0](https://github.com/Bearer/bearer/compare/v0.79.0...v0.80.0) (2018-12-17)

**Note:** Version bump only for package @bearer/templates





<a name="0.79.0"></a>
# [0.79.0](https://github.com/Bearer/bearer/compare/v0.78.0...v0.79.0) (2018-12-14)

**Note:** Version bump only for package @bearer/templates





<a name="0.78.0"></a>
# [0.78.0](https://github.com/Bearer/bearer/compare/v0.77.0...v0.78.0) (2018-12-14)

**Note:** Version bump only for package @bearer/templates





<a name="0.77.0"></a>
# [0.77.0](https://github.com/Bearer/bearer/compare/v0.76.0...v0.77.0) (2018-12-13)

**Note:** Version bump only for package @bearer/templates





<a name="0.76.0"></a>
# [0.76.0](https://github.com/Bearer/bearer/compare/v0.75.0...v0.76.0) (2018-12-12)

**Note:** Version bump only for package @bearer/templates





<a name="0.75.0"></a>
# [0.75.0](https://github.com/Bearer/bearer/compare/v0.74.7...v0.75.0) (2018-12-12)

**Note:** Version bump only for package @bearer/templates





<a name="0.74.6"></a>
## [0.74.6](https://github.com/Bearer/bearer/compare/v0.74.5...v0.74.6) (2018-12-07)

**Note:** Version bump only for package @bearer/templates





<a name="0.74.0"></a>
# [0.74.0](https://github.com/Bearer/bearer/compare/v0.73.2...v0.74.0) (2018-12-01)

**Note:** Version bump only for package @bearer/templates





<a name="0.73.0"></a>
# [0.73.0](https://github.com/Bearer/bearer/compare/v0.72.1...v0.73.0) (2018-11-16)

**Note:** Version bump only for package @bearer/templates





<a name="0.72.1"></a>
## [0.72.1](https://github.com/Bearer/bearer/compare/v0.72.0...v0.72.1) (2018-11-15)

**Note:** Version bump only for package @bearer/templates





<a name="0.72.0"></a>
# [0.72.0](https://github.com/Bearer/bearer/compare/v0.71.4...v0.72.0) (2018-11-13)

**Note:** Version bump only for package @bearer/templates





<a name="0.71.4"></a>
## [0.71.4](https://github.com/Bearer/bearer/compare/v0.71.3...v0.71.4) (2018-11-09)

**Note:** Version bump only for package @bearer/templates





<a name="0.71.0"></a>
# [0.71.0](https://github.com/Bearer/bearer/compare/v0.70.3...v0.71.0) (2018-10-24)

**Note:** Version bump only for package @bearer/templates





<a name="0.70.2"></a>
## [0.70.2](https://github.com/Bearer/bearer/compare/v0.70.1...v0.70.2) (2018-10-18)

**Note:** Version bump only for package @bearer/templates





<a name="0.70.1"></a>
## [0.70.1](https://github.com/Bearer/bearer/compare/v0.70.0...v0.70.1) (2018-10-18)

**Note:** Version bump only for package @bearer/templates





<a name="0.70.0"></a>
# [0.70.0](https://github.com/Bearer/bearer/compare/v0.69.0...v0.70.0) (2018-10-18)

**Note:** Version bump only for package @bearer/templates





<a name="0.69.0"></a>
# [0.69.0](https://github.com/Bearer/bearer/compare/v0.68.1...v0.69.0) (2018-10-17)

**Note:** Version bump only for package @bearer/templates





<a name="0.68.0"></a>
# [0.68.0](https://github.com/Bearer/bearer/compare/v0.67.6...v0.68.0) (2018-10-16)

**Note:** Version bump only for package @bearer/templates





<a name="0.67.6"></a>
## [0.67.6](https://github.com/Bearer/bearer/compare/v0.67.5...v0.67.6) (2018-10-15)

**Note:** Version bump only for package @bearer/templates





<a name="0.67.5"></a>
## [0.67.5](https://github.com/Bearer/bearer/compare/v0.67.4...v0.67.5) (2018-10-11)

**Note:** Version bump only for package @bearer/templates





<a name="0.67.3"></a>
## [0.67.3](https://github.com/Bearer/bearer/compare/v0.67.2...v0.67.3) (2018-10-10)

**Note:** Version bump only for package @bearer/templates





<a name="0.67.1"></a>
## [0.67.1](https://github.com/Bearer/bearer/compare/v0.67.0...v0.67.1) (2018-10-10)

**Note:** Version bump only for package @bearer/templates





<a name="0.67.0"></a>
# [0.67.0](https://github.com/Bearer/bearer/compare/v0.66.1...v0.67.0) (2018-10-09)

**Note:** Version bump only for package @bearer/templates





<a name="0.66.0"></a>
# [0.66.0](https://github.com/Bearer/bearer/compare/v0.65.1...v0.66.0) (2018-10-09)

**Note:** Version bump only for package @bearer/templates





<a name="0.65.1"></a>
## [0.65.1](https://github.com/Bearer/bearer/compare/v0.65.0...v0.65.1) (2018-10-03)

**Note:** Version bump only for package @bearer/templates





<a name="0.65.0"></a>
# [0.65.0](https://github.com/Bearer/bearer/compare/v0.64.1...v0.65.0) (2018-10-02)




**Note:** Version bump only for package @bearer/templates

<a name="0.64.0"></a>
# [0.64.0](https://github.com/Bearer/bearer/compare/v0.63.0...v0.64.0) (2018-09-28)

**Note:** Version bump only for package @bearer/templates





<a name="0.63.0"></a>
# [0.63.0](https://github.com/Bearer/bearer/compare/v0.62.1...v0.63.0) (2018-09-26)

**Note:** Version bump only for package @bearer/templates





<a name="0.62.0"></a>
# [0.62.0](https://github.com/Bearer/bearer/compare/v0.61.0...v0.62.0) (2018-09-25)

**Note:** Version bump only for package @bearer/templates





<a name="0.61.0"></a>
# [0.61.0](https://github.com/Bearer/bearer/compare/v0.60.13...v0.61.0) (2018-09-18)

**Note:** Version bump only for package @bearer/templates





<a name="0.60.13"></a>
## [0.60.13](https://github.com/Bearer/bearer/compare/v0.60.12...v0.60.13) (2018-09-14)

**Note:** Version bump only for package @bearer/templates





<a name="0.60.12"></a>
## [0.60.12](https://github.com/Bearer/bearer/compare/v0.60.11...v0.60.12) (2018-09-13)




**Note:** Version bump only for package @bearer/templates

<a name="0.60.11"></a>
## [0.60.11](https://github.com/Bearer/bearer/compare/v0.60.10...v0.60.11) (2018-09-13)




**Note:** Version bump only for package @bearer/templates

<a name="0.60.10"></a>
## [0.60.10](https://github.com/Bearer/bearer/compare/v0.60.9...v0.60.10) (2018-09-13)




**Note:** Version bump only for package @bearer/templates

<a name="0.60.9"></a>
## [0.60.9](https://github.com/Bearer/bearer/compare/v0.60.8...v0.60.9) (2018-09-13)




**Note:** Version bump only for package @bearer/templates

<a name="0.60.8"></a>
## [0.60.8](https://github.com/Bearer/bearer/compare/v0.60.7...v0.60.8) (2018-09-13)




**Note:** Version bump only for package @bearer/templates

<a name="0.60.7"></a>
## [0.60.7](https://github.com/Bearer/bearer/compare/v0.60.6...v0.60.7) (2018-09-13)




**Note:** Version bump only for package @bearer/templates

<a name="0.60.6"></a>
## [0.60.6](https://github.com/Bearer/bearer/compare/v0.60.5...v0.60.6) (2018-09-12)




**Note:** Version bump only for package @bearer/templates

<a name="0.60.5"></a>
## [0.60.5](https://github.com/Bearer/bearer/compare/v0.60.4...v0.60.5) (2018-09-12)




**Note:** Version bump only for package @bearer/templates

<a name="0.60.0"></a>
# [0.60.0](https://github.com/Bearer/bearer/compare/v0.59.1...v0.60.0) (2018-09-07)

**Note:** Version bump only for package @bearer/templates





<a name="0.59.0"></a>
# [0.59.0](https://github.com/Bearer/bearer/compare/v0.58.8...v0.59.0) (2018-09-03)

**Note:** Version bump only for package @bearer/templates





<a name="0.58.7"></a>
## [0.58.7](https://github.com/Bearer/bearer/compare/v0.58.6...v0.58.7) (2018-08-28)

**Note:** Version bump only for package @bearer/templates





<a name="0.58.6"></a>
## [0.58.6](https://github.com/Bearer/bearer/compare/v0.58.5...v0.58.6) (2018-08-28)

**Note:** Version bump only for package @bearer/templates





<a name="0.58.5"></a>
## [0.58.5](https://github.com/Bearer/bearer/compare/v0.57.3...v0.58.5) (2018-08-27)

**Note:** Version bump only for package @bearer/templates





<a name="0.57.2"></a>
## [0.57.2](https://github.com/Bearer/bearer/compare/v0.57.1...v0.57.2) (2018-08-27)

**Note:** Version bump only for package @bearer/templates





<a name="0.57.0"></a>
# [0.57.0](https://github.com/Bearer/bearer/compare/v0.56.3...v0.57.0) (2018-08-27)

**Note:** Version bump only for package @bearer/templates





<a name="0.56.0"></a>
# [0.56.0](https://github.com/Bearer/bearer/compare/v0.55.0...v0.56.0) (2018-08-22)


### Features

* **cli:** generate command rewrite ([#287](https://github.com/Bearer/bearer/issues/287)) ([2e91da7](https://github.com/Bearer/bearer/commit/2e91da7))




<a name="0.53.0"></a>
# [0.53.0](https://github.com/Bearer/bearer/compare/v0.52.0...v0.53.0) (2018-08-14)


### Bug Fixes

* **ui:** change LICENSE to MIT ([d90917d](https://github.com/Bearer/bearer/commit/d90917d))




<a name="0.50.0"></a>
# [0.50.0](https://github.com/Bearer/bearer/compare/v0.48.5...v0.50.0) (2018-08-13)


### Bug Fixes

* **templates:** wrong callback call ([aa99089](https://github.com/Bearer/bearer/commit/aa99089))




<a name="0.48.4"></a>
## [0.48.4](https://github.com/Bearer/bearer/compare/v0.48.3...v0.48.4) (2018-08-10)

**Note:** Version bump only for package @bearer/templates





<a name="0.43.1"></a>
## [0.43.1](https://github.com/Bearer/bearer/compare/v0.43.0...v0.43.1) (2018-08-10)

**Note:** Version bump only for package @bearer/templates





<a name="0.42.0"></a>
# [0.42.0](https://github.com/Bearer/bearer/compare/v0.41.9...v0.42.0) (2018-08-10)

**Note:** Version bump only for package @bearer/templates





<a name="0.41.1"></a>
## [0.41.1](https://github.com/Bearer/bearer/compare/v0.41.0...v0.41.1) (2018-08-08)

**Note:** Version bump only for package @bearer/templates





<a name="0.41.0"></a>
# [0.41.0](https://github.com/Bearer/bearer/compare/v0.40.0...v0.41.0) (2018-08-08)

**Note:** Version bump only for package @bearer/templates





<a name="0.39.0"></a>
# [0.39.0](https://github.com/Bearer/bearer/compare/v0.38.1...v0.39.0) (2018-08-08)


### Features

* **core:** everything as post ([#234](https://github.com/Bearer/bearer/issues/234)) ([ae67130](https://github.com/Bearer/bearer/commit/ae67130))





<a name="0.37.2"></a>
## [0.37.2](https://github.com/Bearer/bearer/compare/v0.37.0...v0.37.2) (2018-08-08)

**Note:** Version bump only for package @bearer/templates





<a name="0.37.1"></a>
## [0.37.1](https://github.com/Bearer/bearer/compare/v0.37.0...v0.37.1) (2018-08-08)

**Note:** Version bump only for package @bearer/templates





<a name="0.37.0"></a>
# [0.37.0](https://github.com/Bearer/bearer/compare/v0.36.3...v0.37.0) (2018-08-08)

**Note:** Version bump only for package @bearer/templates





<a name="0.35.0"></a>
# [0.35.0](https://github.com/Bearer/bearer/compare/v0.34.2...v0.35.0) (2018-08-07)


### Bug Fixes

* **cli:** use correct intent type name ([2d5ffa4](https://github.com/Bearer/bearer/commit/2d5ffa4))





<a name="0.34.0"></a>
# [0.34.0](https://github.com/Bearer/bearer/compare/v0.33.0...v0.34.0) (2018-08-03)

**Note:** Version bump only for package @bearer/templates





<a name="0.24.0"></a>
# [0.24.0](https://github.com/Bearer/bearer/compare/v0.23.2...v0.24.0) (2018-07-25)


### Features

* **generator:** Separate TContext per AuthType ([#152](https://github.com/Bearer/bearer/issues/152)) ([b194c23](https://github.com/Bearer/bearer/commit/b194c23))




<a name="0.20.0"></a>
# [0.20.0](https://github.com/Bearer/bearer/compare/v0.19.1...v0.20.0) (2018-07-20)


### Features

* **templates:** Add BasicAuth and update Client ([#120](https://github.com/Bearer/bearer/issues/120)) ([afc0192](https://github.com/Bearer/bearer/commit/afc0192))




<a name="0.18.0"></a>
# [0.18.0](https://github.com/Bearer/bearer/compare/v0.17.5...v0.18.0) (2018-07-19)


### Features

* **generator:** improves client.ts ([#107](https://github.com/Bearer/bearer/issues/107)) ([5c5ffcc](https://github.com/Bearer/bearer/commit/5c5ffcc))
