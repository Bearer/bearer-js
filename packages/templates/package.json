{
  "name": "@bearer/templates",
  "version": "0.45.0",
  "description": "Bearer templates",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "author": "Bearer",
  "license": "UNLICENSED",
  "files": [
    "dist/"
  ],
  "scripts": {
    "start": "yarn build",
    "build": "tsc -p tsconfig.json",
    "prepublish": "yarn build",
    "prepublishOnly": "yarn build",
    "test": "jest"
  },
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "<rootDir>/preprocessor.js"
    },
    "testMatch": [
      "**/__tests__/*.+(ts|tsx|js)"
    ],
    "setupFiles": [
      "./jest.setup.js"
    ]
  },
  "repository": {
    "type": "git",
    "url": "github:Bearer/bearer"
  },
  "dependencies": {
    "typescript": "^2.9.2"
  },
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@types/node": "^10.5.2",
    "jest": "^23.4.0",
    "ts-jest": "^23.0.0"
  }
}
