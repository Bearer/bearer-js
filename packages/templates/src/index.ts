import Authentication from "@bearer/types/lib/Authentications";

import apikey from './apikey'
import basicauth from './basicauth'
import noauth from './noauth'
import oauth2 from './oauth2'

type  TTemplate = {
  SaveState: string,
  RetrieveState: string,
  FetchData: string
}

export const templates: Record<Authentication, TTemplate> = {
  [Authentication.OAuth2]: oauth2,
  [Authentication.ApiKey]: apikey,
  [Authentication.Basic]: basicauth,
  [Authentication.NoAuth]: noauth,
}

export default templates
