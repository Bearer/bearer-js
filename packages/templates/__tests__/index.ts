import templates from '../src/index'
import Authentications from "@bearer/types/lib/Authentications";

describe('Templates', () => {
  Object.values(Authentications).forEach((key) => {
    const intent = templates[key]

    describe(key, () => {
      it('has SaveState', () => {
        expect(intent.SaveState).toMatchSnapshot()
      })

      it('has RetrieveState', () => {
        expect(intent.RetrieveState).toMatchSnapshot()
      })

      it('has FetchData', () => {
        expect(intent.FetchData).toMatchSnapshot()
      })
    })
  })
});