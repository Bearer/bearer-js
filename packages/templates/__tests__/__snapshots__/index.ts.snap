// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates APIKEY has FetchData 1`] = `
"
  static action(context: TAPIKEYAuthContext, params: any, body: any, callback: TFetchDataCallback) {
    //... your code goes here
    // use the client defined in client.ts to fetch real object like that:
    // Client(context.authAccess.apiKey).get('/people')
    //   .then(({ data }) => {
    //     callback({ data })
    //   })
    //   .catch((error) => {
    //     callback({ error: error.toString() })
    //   })
    callback({ data: []})
  }
  "
`;

exports[`Templates APIKEY has RetrieveState 1`] = `
"
  static action(_context: TAPIKEYAuthContext, _params: any, state, callback: TRetrieveStateCallback) {
    callback({ data: state })
  }
  "
`;

exports[`Templates APIKEY has SaveState 1`] = `
"
  static action(
    _context: TAPIKEYAuthContext,
    _params: any,
    body: any,
    state: any,
    callback: TSaveStateCallback
  ): void {
    const { item: { name } } = body
    const { items = [] }: any = state
    const newItem: any = { name }

    callback({
      state: {
        ...state,
        items: [...items, newItem]
      },
      data: [...items, newItem]
    })
  }
  "
`;

exports[`Templates BASIC has FetchData 1`] = `
"
  static action(context: TBASICAuthContext, params: any, body: any, callback: TFetchDataCallback) {
    //... your code goes here
    // use the client defined in client.ts to fetch real object like that:
    // Client(
    //   context.authAccess.username,
    //   context.authAccess.password
    // ).get('/people').then(({ data }) => {
    //     callback({ data })
    //   })
    //   .catch((error) => {
    //     callback({ error: error.toString() })
    //   })
    callback({ data: []})
  }
  "
`;

exports[`Templates BASIC has RetrieveState 1`] = `
"
  static action(_context: TBASICAuthContext, _params: any, state, callback: TRetrieveStateCallback) {
    callback({ data: state })
  }
  "
`;

exports[`Templates BASIC has SaveState 1`] = `
"
  static action(
    _context: TBASICAuthContext,
    _params: any,
    body: any,
    state: any,
    callback: TSaveStateCallback
  ): void {
    const { item: { name } } = body
    const { items = [] }: any = state
    const newItem: any = { name }

    callback({
      state: {
        ...state,
        items: [...items, newItem]
      },
      data: [...items, newItem]
    })
  }
  "
`;

exports[`Templates NONE has FetchData 1`] = `
"
  static action(context: TNONEAuthContext, params: any, body: any, callback: TFetchDataCallback) {
    //... your code goes here
    callback({ data: []})
  }"
`;

exports[`Templates NONE has RetrieveState 1`] = `
"
  static action(_context: TNONEAuthContext, _params: any, state, callback: TRetrieveStateCallback) {
    callback({ data: state })
  }
  "
`;

exports[`Templates NONE has SaveState 1`] = `
"
  static action(
    _context: TNONEAuthContext,
    _params: any,
    body: any,
    state: any,
    callback: TSaveStateCallback
  ): void {
    const { item: { name } } = body
    const { items = [] }: any = state
    const newItem: any = { name }

    callback({
      state: {
        ...state,
        items: [...items, newItem]
      },
      data: [...items, newItem]
    })
  }
  "
`;

exports[`Templates OAUTH2 has FetchData 1`] = `
"
  static action(context: TOAUTH2AuthContext, params: any, body: any, callback: TFetchDataCallback) {
    //... your code goes here
    // use the client defined in client.ts to fetch real object like that:
    // Client(context.authAccess.accessToken).get('/people').then(({ data }) => {
    //     callback({ data })
    //   })
    //   .catch((error) => {
    //     callback({ error: error.toString() })
    //   })
    callback({ data: []})
  }"
`;

exports[`Templates OAUTH2 has RetrieveState 1`] = `
"
  static action(_context: TOAUTH2AuthContext, _params: any, state, callback: TRetrieveStateCallback) {
    callback({ data: state })
  }
  "
`;

exports[`Templates OAUTH2 has SaveState 1`] = `
"
  static action(
    _context: TOAUTH2AuthContext,
    _params: any,
    body: any,
    state: any,
    callback: TSaveStateCallback
  ): void {
    const { item: { name } } = body
    const { items = [] }: any = state
    const newItem: any = { name }

    callback({
      state: {
        ...state,
        items: [...items, newItem]
      },
      data: [...items, newItem]
    })
  }
  "
`;
